src/Algebra/Construct/Initial.agda

```diff
@@ -0,0 +1,92 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Instances of algebraic structures where the carrier is ⊥.
+-- In mathematics, this is usually called 0.
+--
+-- From monoids up, these are zero-objects – i.e, the initial
+-- object is the terminal object in the relevant category.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Level using (Level)
+
+module Algebra.Construct.Initial {c ℓ : Level} where
+
+open import Algebra.Bundles
+  using (Magma; Semigroup; Band)
+open import Algebra.Bundles.Raw
+  using (RawMagma)
+open import Algebra.Core using (Op₂)
+open import Algebra.Definitions using (Congruent₂)
+open import Algebra.Structures using (IsMagma; IsSemigroup; IsBand)
+open import Data.Empty.Polymorphic
+open import Relation.Binary using (Rel; Reflexive; Symmetric; Transitive; IsEquivalence)
+
+
+------------------------------------------------------------------------
+-- re-export those algebras which are also terminal
+
+open import Algebra.Construct.Terminal {c} {ℓ} public
+  hiding (rawMagma; magma; semigroup; band)
+
+------------------------------------------------------------------------
+-- gather all the functionality in one place
+
+private module ℤero where
+
+  infixl 7 _∙_
+  infix  4 _≈_
+  Carrier : Set c
+  _≈_     : Rel Carrier ℓ
+  _∙_     : Op₂ Carrier
+
+  Carrier = ⊥
+  _≈_ ()
+  _∙_ ()
+
+  refl : Reflexive _≈_
+  refl {x = ()}
+
+  sym : Symmetric _≈_
+  sym {x = ()}
+
+  trans : Transitive _≈_
+  trans {i = ()}
+
+  isEquivalence : IsEquivalence _≈_
```

Comment:
It's often quite useful to have the structures as well as the bundles exported publicly. Can we move the structures out of the private module?

Reply:
Oh, OK...

Reply:
Just in the middle of trying to do this, and thinking: 
could I achieve the same effect by suitably modifying the `open ℤero` declaration?

Reply:
Yes you could. By duplicating it and having one with a `public using (X)` and one with a `hiding (X)`.

Reply:
Hmm... (I think) I see. Have left thinsg as they were: definitions inside the (private) module; structures/bundles outside. Hope that's (still) OK?!